src/server/editorServices.ts

```diff
@@ -1037,6 +1053,97 @@ namespace ts.server {
             return undefined;
         }
 
+        acquireTypingForJs(project: Project) {
+            if (!project) { return; }
+
+            let tsd: any;
+            try {
+                tsd = require("tsd");
+            }
+            catch (e) {
+                this.log("Error requiring the tsd module");
+                return;
+            }
+
+            const cachePath = project.isConfiguredProject()
+                ? toPath(getDirectoryPath(project.projectFilename), project.projectFilename, getCanonicalFileName)
+                : globalCachePath;
+
+            // For inferred project, we always enable the auto typing discover feature, therefore the typingOptions
+            // has a default value
+            const typingOptions: TypingOptions =
+                project.projectOptions ? project.projectOptions.typingOptions : { enableAutoDiscovery: true };
+            const compilerOptions = project.projectOptions ? project.projectOptions.compilerOptions : undefined;
+
+            const { cachedTypingPaths, newTypingNames } = ts.JsTyping.discoverTypings(
+                sys,
+                project.getFileNames(),
+                globalCachePath,
+                cachePath,
+                typingOptions,
+                compilerOptions
+            );
+
+            // Bail out when the autoTyping is disabled

```

Comment:
I was trying to say "Bail out when no actions needed", will update to make it clearer.